thunder/dynamo/utils.py

```diff
@@ -306,6 +328,14 @@ def is_node_supported_by_thunder(node: torch.fx.Node) -> tuple[bool, SplitReason
         )
         return False, split_reason
 
+    # If the operation is higher order function for checkpointing, check whether the submodule is supported by Thunder
```

Comment:
It's better to explain why this if-statement is needed and not what it does.
```suggestion
    # The checkpointed function must be fully supported by Thunder
```